pallets/avn-transaction-payment/src/tests/adjustment_fee_tests.rs

```diff
@@ -0,0 +1,582 @@
+                Some(inner)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>()
+        .len() >
+        0
+}
+
+/// Rolls desired block number of times.
+pub(crate) fn roll_blocks(n: u64) {
+    let mut block = System::block_number();
+    let target_block = block + n;
+    while block < target_block {
+        block = roll_one_block();
+    }
+}
+
+mod adjustment_fee_tests {
+    use super::*;
+
+    mod fees_are_paid_correctly_when {
+        use super::*;
+
+        mod call_is_set_without_adjustment_type {
+            use super::*;
+            use crate::mock::AvnTransactionPayment;
+
+            #[test]
+            fn and_valid_fixed_fee() {
+                new_test_ext().execute_with(|| {
+                    let sender = to_acc_id(1u64);
+                    set_initial_sender_balance(&sender);
+
+                    let config = AdjustmentInput::<TestRuntime> {
+                        fee_type: FeeType::FixedFee(FixedFeeConfig { fee: FIXED_FEE }),
+                        adjustment_type: AdjustmentType::None,
+                    };
+
+                    assert_ok!(AvnTransactionPayment::set_known_sender(
+                        RuntimeOrigin::root(),
+                        sender,
+                        config,
+                    ));
+                    assert_eq!(AvnTransactionPayment::is_known_sender(sender), true);
+
+                    pay_gas_and_call_remark(&sender);
+                    assert_eq!(INITIAL_SENDER_BALANCE - FIXED_FEE, Balances::free_balance(sender));
+
+                    assert!(event_emitted(&mock::RuntimeEvent::AvnTransactionPayment(
+                        crate::Event::<TestRuntime>::AdjustedTransactionFeePaid {
+                            who: sender,
+                            fee: FIXED_FEE,
+                        }
+                    )));
+                })
+            }
+
+            #[test]
+            fn and_fixed_fee_higher_than_base_fee() {
+                new_test_ext().execute_with(|| {
+                    let sender = to_acc_id(1u64);
+                    set_initial_sender_balance(&sender);
+
+                    let higher_fixed_fee = u128::from(BASE_FEE + 1);
+                    let config = AdjustmentInput::<TestRuntime> {
+                        fee_type: FeeType::FixedFee(FixedFeeConfig { fee: higher_fixed_fee }),
+                        adjustment_type: AdjustmentType::None,
+                    };
+
+                    assert_ok!(AvnTransactionPayment::set_known_sender(
+                        RuntimeOrigin::root(),
+                        sender,
+                        config,
+                    ));
+                    assert_eq!(AvnTransactionPayment::is_known_sender(sender), true);
+
+                    pay_gas_and_call_remark(&sender);
+                    assert_eq!(INITIAL_SENDER_BALANCE - u128::from(BASE_FEE), Balances::free_balance(sender));
```

Comment:
I'm confused here. So what is the config in this case? Is it a discount whose value is the `base_fee + 1`? But then, shouldn't we have not paid `0`, instead of paying `base_fee`?

Reply:
The value we pass is not a discount, it is a fixed fee. Meaning the value we pass in will be the value we pay for the fee.

If we provide the fixed fee of 15, we don't charge anything more that the chain base fee. 
We just do a discount if the fixed fee is lower than our base fee.